conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/MockEnclaveEnvironment.kt

```diff
@@ -103,6 +103,18 @@ class MockEnclaveEnvironment(
         return report
     }
 
+    override fun getSignedQuote(
+        quotingEnclaveInfo: ByteCursor<SgxTargetInfo>?,
+        reportData: ByteCursor<SgxReportData>?
+    ): ByteCursor<SgxSignedQuote> {
+        //  Note that the quotingEnclaveInfo passed in is empty.
+        //  We could have directly created an empty quotingEnclaveInfo here, 
+        //    but we want to keep the same mechanism we have in the Graal/Native Image flow, as this is
+        //    what Mock mode is trying to emulate.
```

Comment:
Not necessarily, especially now that Gramine does it differently to GraalVM. My immediate thought here is that we now have the wrong abstraction and need to tweak it.

Reply:
I have noticed that in Mock mode we pass an empty `quotingEnclaveInfo`.
What do you mean by "not necessarily"? I haven't changed mock mode, just refactored functions to follow the common interface.

Reply:
>What do you mean by "not necessarily"?

That mock mode should emulate GraalVM, at least now that we have Gramine doing differently.

Reply:
Sorry, I don't understand the point.
I have just added a comment here, there are no functional changes of the mock mode flow, just the implementation.
Mock mode WAS emulating Graal VM and it IS emulating Graal VM way of signing the quote (or better that ping-pong host-enclave mechanism). What do I need to fix?
Or did I change anything without noticing?

Reply:
I understand you've not change anything with regards to mock mode behaviour. I guess what I'm saying is, now that Gramine has been introduced and it does things differently, shouldn't mock mode also not be changed to do this differently, or rather more simply? There's no reason now why mock mode should emulate graalvm. But I guess that is out of scope of this PR and can be done later? The reason I think it could be done here is if we also redesigned the host's attestation abstraction in this PR then we can simply mock mode attestaion as well.

I guess this can be all done in a separate PR. I guess add a TODO about changing the abstaction and simplifying mock mode attestsation.

Reply:
Ah ok yes, but then what should Mock mode emulate? Graal or Gramine or both? I just thought that emulating the most complex flow was more relevant. If not, it would imply that we could have simplified things before, then, correct?

Reply:
Since Graal and Gramine do things slightly different from each other, I think the solution for mock is always to do things in its own way that makes most sense. This should probably be done as part of fixing the attestation abstraction, which can occur in a followup ticket.

Reply:
Is this ok now? If so, please mark as resolved